### ISSUE TITLE
hard to place a rapido app in a site

### ISSUE THREAD

**Issue by djay** [plomino/rapido.plone#9]
# User problem

Lets I have an function that I want to be in /voting and fill up the main content and subtree of that part of the site.

# Options

## diazo path conditional (current)

Currently you can do that my using diazo with a path expression. But this is brittle, as I might change that folder id.

## tinymce style (current)
Alternatively I could create a tinymce style, create a page with any id and insert the style into my page.

## rapido tile (current)
Make the page a mosaic page, insert a rapido tile and select the block.

## displayview

- Add a new setting inside a block settings to "make available as Display menu item".
- User gives it a displayview title and perhaps a set of content types. 
- user creates a page/folder and selects "my app" from the display view.

**Comment by djay** [comment]
@ebrehault I think I like the "new tile type" idea. ideally it would be nice if rapido could take advantage of DX schema so that you didn't have to write a lot of code handle simple settings. maybe if could use z3cform fields as elements, and perhaps leave the html blank?

**Comment by ebrehault** [comment]
Yes, being able to turn a given block into a tile would be very nice, and that's something I plan to do.

Note: it is certainly possible to use dexterity and/or z3c.form in rapido, but the resulting code would be very complex, and I want rapido to keep simple. Nevertheless, I also think using the form tooling of Plone in Rapido is interesting, and my approach in a recent project was just to use collective.easyform: the user manages form fields with EasyForm, and my block just call the form rendering and then process/store the data. That's very simple and it avoids to bring the dx/z3c.form complexity into rapido directly.

**Comment by djay** [comment]
Interesting approach. Means you global theme based code relying on a form in a specific location which doesn't seem to logically make sense. But I guess its not that much different to hooking rapido into a specific path using diazo. Not sure if its possible to get a collective.easyform to use rapido to handle submission which would make more logical sense. 
In terms of rapido using z3cform widgets directly I think its a nice to have but thinking about it more, it doesn't solve a big problem. It just makes it quicker to create certain kinds of forms and take advantage patternslib more easily. For example if you want a mutltiselect, or autocomplete. I'm not sure it makes sense for rapido to make its own versions of those?

**Comment by djay** [comment]
I just added in the jbot replacement option.
